code/extensions/che-terminal/src/extension.ts

```diff
@@ -127,3 +128,29 @@ export class MachineExecPTY implements vscode.Pseudoterminal {
 
 export function deactivate(): void {
 }
+
+/**
+ * Provides current working directory for a terminal creation:
+ *
+ * @returns:
+ * - value of the PROJECTS_ROOT env variable if there is no any workspace folder
+ * - path to a folder if there is only one workspace folder
+ * - IDE proposes an user to select a folder if there is more then 1 workspace folder,
+ * current function returns path to the selected folder    
+ */
+async function getCurrentWorkingDirectory(): Promise<string | undefined> {
+	const folders = vscode.workspace.workspaceFolders;
+	if (folders === undefined || folders.length < 1) {
+		return process.env.PROJECTS_ROOT as string;
+	}
+
+	if (folders.length === 1) {
+		return folders[0].uri.path
+	}
+
+	const options = {
+		placeHolder: "Select current working directory for new terminal"
```

Comment:
I would change the message to
```suggestion
		placeHolder: "Select a working directory for the new terminal"
```

Reply:
I prefer to leave it as is.
VS Code uses the same message

Reply:
Please see, for example, here  https://github.com/microsoft/vscode/blob/38623f21e7de9d8801a2c1eb7eeda514398a3e77/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L1810